### ISSUE TITLE
Xiaomi cant connect to bluetooth

### ISSUE THREAD

**Issue by leifj79** [nisargjhaveri/AAWirelessDongle#14]
Have an xiaomi device and i cant connect to the rpi 2 zero by bluetooth using the latest code! 
Have tested BluemediaGER fork and there i can connect and it shows up correctly but then it cant connect to my cars android auto. 
The bluetooth doesn't show up unless i tell my phone to not show names on bluetooth and then i can see the mac adress of the device but pairing fails!

**Comment by nu00** [comment]
Try to build the project from source, with #7 have been made some changes in VID and PID of the USB mode that might help you with your problem

**Comment by leifj79** [comment]
Im building from ca7859b and will be back with logs

**Comment by nisargjhaveri** [comment]
You can also try the [latest build on main from Github Actions](https://github.com/nisargjhaveri/AAWirelessDongle/actions).

**Comment by nisargjhaveri** [comment]
The logs suggests that the headunit is not recognising and not sending the accessory start request. I'm out of ideas on why this would happen if Android Auto works in the car, as we're very close to what Pixel devices report via USB.

**Comment by leifj79** [comment]
The head unit shows it's connected to Android Auto

**Comment by leifj79** [comment]
I guess its my head unit that is to blame, i have to choose between android auto and mirrorlink and i think that is the issue :(
haven't found any way to disable mirrorlink in the car, but newer models dosnt have mirrorlink and it seems fine!

**Comment by nisargjhaveri** [comment]
> I guess its my head unit that is to blame, i have to choose between android auto and mirrorlink and i think that is the issue :(

It is technically supposed to work once you select Android Auto. I guess #2 also discusses about a headunit that has similar options for Android Auto and MirrorLink.

**Comment by leifj79** [comment]
Yeah the BT starts and my phone tried to connect but doesn't connect before the head unit resets

**Comment by nisargjhaveri** [comment]
Right, you can maybe try out the suggestion here: https://github.com/nisargjhaveri/AAWirelessDongle/issues/2#issuecomment-1783434263

Though, this looks like a common issue. Let me try to make a permenant change to fix the timeouts.

**Comment by leifj79** [comment]
Tested BluemediaGER version and it works, now I'm going wireless with my Passat 🥳

**Comment by leifj79** [comment]
Works with suggested solution

**Comment by leifj79** [comment]
Had some new problems but I built from the tree optimistic_phone_connection and my wolksvagen accepts it without any issues! Awsome work!
